### PR TITLE
cmd/snap: coldplug auto-import assertions from all removable devices

### PR DESCRIPTION
**Pull request by bboozzoo**
Add support for importing assertions from all removable devices in coldplug
scenario. The snap auto-import --mount invocations from udev triggers that run
before snapd is seeded in first boot will fail because snap command is not yet
available. The coldplug scenario allows us to import assertions from all
removable devices when snap.auto-import is explicitly started.

### CODE REVIEW

File: cmd/snap/cmd_auto_import.go [hunk 1]
```diff
@@ -263,7 +267,44 @@ func (x *cmdAutoImport) autoAddUsers() error {
...
+
+		for _, partAttr := range partitionAttrs {
+			val, err := ioutil.ReadFile(partAttr)
+			if err != nil || string(val) != "1\n" {
+				// non partition?
+				continue
+			}
+			pdev := filepath.Base(filepath.Dir(partAttr))
+			removableDevices = append(removableDevices, fmt.Sprintf("/dev/%s", pdev))
+			// hasPartitions = true
```
Comment (anonymouse64) [inline cmd/snap/cmd_auto_import.go:303]: leftover code?

File: cmd/snap/cmd_auto_import_test.go [hunk 2]
```diff
@@ -321,3 +323,111 @@ func (s *SnapSuite) TestAutoImportUnhappyInInstallMode(c *C) {
...
+	})
+
+	// do not mock mountinfo contents, we just want to observe whether we
+	// try to mount and umount the right stuff
+
+	_, err := snap.Parser(snap.Client()).ParseArgs([]string{"auto-import"})
+	c.Assert(err, IsNil)
+	c.Check(s.Stdout(), Equals, "")
+	c.Check(s.Stderr(), Equals, "")
+	c.Check(mountCmd.Calls(), DeepEquals, [][]string{
```
Comment (anonymouse64) [inline cmd/snap/cmd_auto_import_test.go:379]: Glob() does not sort, so either you should sort the output from Glob(), or sort the output here. I have a slight preference to sort the Glob() output in the code.

Comment (bboozzoo) [reply cmd/snap/cmd_auto_import_test.go:379]: Fair enough, it's an implementation detail of Glob that's not mentioned in the docs, so making it explicit makes sense.

**Comment by bboozzoo** [comment]
The failures are unrelated. 
```
2020-05-11T07:32:21.2929426Z 2020-05-11 07:32:21 Failed project prepare: 6
2020-05-11T07:32:21.2929717Z     - google:ubuntu-14.04-64:project
2020-05-11T07:32:21.2929999Z     - google:ubuntu-14.04-64:project
2020-05-11T07:32:21.2930302Z     - google:ubuntu-14.04-64:project
2020-05-11T07:32:21.2930576Z     - google:ubuntu-14.04-64:project
2020-05-11T07:32:21.2930837Z     - google:ubuntu-14.04-64:project
2020-05-11T07:32:21.2931110Z     - google:ubuntu-14.04-64:project

2020-05-11 08:04:59 Failed tasks: 1
    - google:ubuntu-16.04-64:tests/main/session-tool-support
```

**Pull request merged**